src/hops/hops.py

```diff
@@ -33,67 +37,52 @@ def get_hops():
         "per_page": hops.per_page
     }
     
-    return jsonify(response), 200
+    return jsonify(response), HTTP_200_OK
 
-@app.route('/hops/', methods=['POST'])
+@hops.post('/')
+@jwt_required()
 def add_new_hop():
     new_hop = Hop(**request.json)
     db.session.add(new_hop)
     db.session.commit()
-    return jsonify(new_hop.as_dict()), 201
+    return jsonify(new_hop.as_dict()), HTTP_201_CREATED
 
-@app.route('/hops/<int:id>', methods=['PUT'])
+@hops.put('/<int:id>/')
+@jwt_required()
 def update_hop(id):
     hop = Hop.query.get(id)
+    
     try:
         for key, value in request.json.items():
             setattr(hop, key, value)
         db.session.commit()
-        return jsonify(hop.as_dict()), 202
+        return jsonify(hop.as_dict()), HTTP_202_ACCEPTED
 
     except IntegrityError:
```

Comment:
Which line can throw `IntegrityError`? Remember to wrap a minimal number of lines of code in the `try:` clause.

Reply:
The name position in my Hop model has a unique value. If I good understand that, If I try to change this unique value the Integrity Error will rise. That was the reason for using it in this case.

Reply:
> If I good understand that, If I try to change this unique value the Integrity Error will rise.
Changing the column which is marked as unique would not raise an error. Changing it in such a way that it would not be unique anymore - so the hop name value would have a duplicate - would raise.


The Idea was to wrap only `db.session.commit()` in a `try:` clause. Remember that you can still use: `else:` clause for a successful case (no exception thrown).

```python
    for key, value in request.json.items():
        setattr(hop, key, value)
    try:
        db.session.commit()
    except IntegrityError:
        ...
    else:
        return jsonify(hop.as_dict()), HTTP_202_ACCEPTED
```

Reply:
Good advice! Already fixed!